### PR TITLE
Copy empty dir integration tests

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Copy Empty directory integration tests
1. TestCopyEmptyDirectoryInNonEmptyDirectory
2. TestCopyEmptyDirectoryInEmptyDirectory
3. TestCopyEmptyDirectoryInNonExistingDirectory

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Run all the tests for testbucket and mountedDirectory flags.(status:SUCCESS)
11. Unit tests - NA
12. Integration tests - Run all the tests for testbucket and mountedDirectory flags.(status:SUCCESS)

### CODE REVIEW

File: tools/integration_tests/operations/copy_dir_test.go [hunk 1]
```diff
@@ -220,3 +219,147 @@ func TestCopyDirectoryInNonEmptyDirectory(t *testing.T) {
 	destSrc := path.Join(destDir, SrcCopyDirectory)
 	checkIfCopiedDirectoryHasCorrectData(destSrc, t)
 }
+
+func checkIfCopiedEmptyDirectoryHasCorrectData(destSrc string, t *testing.T) {
```
Comment (Tulsishah) [inline tools/integration_tests/operations/copy_dir_test.go:223]: Should I rename [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/operations/copy_dir_test.go#L68) to checkIfCopiedNonEmptyDirectoryHasCorrectData ?

Comment (raj-prince) [reply tools/integration_tests/operations/copy_dir_test.go:223]: Empty directory means no data. CorrectData doesn't make sense for empty directory.

Instead we should rename this to - checkIfCopiedEmptyDirectoryHasNoData or similar.

File: tools/integration_tests/operations/copy_dir_test.go [hunk 2]
```diff
@@ -220,3 +219,147 @@ func TestCopyDirectoryInNonEmptyDirectory(t *testing.T) {
...
+	objs, err := os.ReadDir(destDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Check if destCopyDirectory has the correct directory copied.
+	// destNonEmptyCopyDirectory
+	// destNonEmptyCopyDirectory/emptyDirectoryCopyTest           - Dir
+	// destNonEmptyCopyDirectory/subDestCopyDirectory             - Dir
+	if len(objs) != NumberOfObjectsInDestCopyDirectory {
```
Comment (raj-prince) [inline tools/integration_tests/operations/copy_dir_test.go:271]: Yes, we need to change this variable name also `NumberOfObjectsInDestCopyDirectory` -> `NumberOfObjectsInNonEmptyDestCopyDirectory`

File: tools/integration_tests/operations/operations_test.go [hunk 3]
```diff
@@ -74,6 +74,11 @@ const NumberOfObjectsInDirThreeInCreateThreeLevelDirTest = 1
 const PrefixFileInDirThreeInCreateThreeLevelDirTest = "fileInDirThreeInCreateThreeLevelDirTest"
 const FileInDirThreeInCreateThreeLevelDirTest = "fileInDirThreeInCreateThreeLevelDirTest1"
 const ContentInFileInDirThreeInCreateThreeLevelDirTest = "Hello world!!"
+const DestEmptyCopyDirectory = "destEmptyCopyDirectory"
```
Comment (raj-prince) [inline tools/integration_tests/operations/operations_test.go:77]: Can we group the variables based on operation? E.g.

// Read
Read related variable...

// Write
Write related variable...

// Copy
Copy related variable...

**Review by raj-prince — APPROVED** [review]
Minor changes, please incorporate before merging.

**Pull request merged**